test/finally_test.cpp

```diff
@@ -45,6 +50,109 @@ TEST(Finally, Value) {
   EXPECT_EQ(res->second, context.get_thread_id());
 }
 
+TEST(Finally, Ref) {
+  {
+    int a = 0;
+
+    auto sndr = just_from([&a]() -> int& { return a; })
+      | finally(just());
+    using Sndr = decltype(sndr);
+
+    static_assert(std::is_same_v<
+      Sndr::value_types<std::variant, std::tuple>,
+      std::variant<std::tuple<int&>>
+    >);
+    static_assert(std::is_same_v<
+      Sndr::error_types<std::variant>,
+      std::variant<std::exception_ptr>
+    >);
+    static_assert(!Sndr::sends_done);
```

Comment:
I would use `sender_traits` here; are you mimicking existing tests?

It's valid to define a _Sender_ that doesn't have these properties in the _Sender_ itself and, also, I'd _like_ to switch from the `sender_traits` model to the queries-and-environments-and-completion-signatures model of P2300/stdexcept, which'll be easier if everyone relies on `sender_traits`.
```suggestion
     sender_value_types_t<Sndr, std::variant, std::tuple>,
      std::variant<std::tuple<int&>>
    >);
    static_assert(std::is_same_v<
      sender_error_types_t<Sndr, std::variant>,
      std::variant<std::exception_ptr>
    >);
    static_assert(!sender_traits<Sndr>::sends_done);
```